### DISCUSSION TITLE
Version 2 release

### DISCUSSION THREAD

**Discussion by amattar** [elsa-workflows/elsa-core#347] [Q&A]
Just wondering, any plans when version 2 shall be released?

**Comment by brightermatter** [comment]
We were also wondering about the 2.0 timeline, we’d be grateful if anyone can shed some light on this.

**Comment by mohdali** [comment]
I also would like to know the timeline for Elsa 2 release.

**Comment by sfmskywalker** [comment]
Hi everyone, thank you for your interest in the V2 release!

Unfortunately I don't have a satisfying answer. Due to other priorities, I had to put a hold on working on V2 for the time being. I'm hopeful I'll be able to pickup the pace in a few months from now, but then it might take a few more months to finish.
So realistically, I wouldn't expect V2 to be ready before the end of this year unless I'm very lucky and I get to shift focus on this earlier.

Main pieces left before V2 is ready to use:

1. Serialization needs to be fixed. Right now it's a bit of a mess.
1. Workflow Server (with REST & GraphQL endpoints to manage workflows)
1. Designer & Dashboard updates.

Originally I had planned to do a complete retake on the designer + dashboard, but I'm realizing that maybe I should do that as part of a follow-up update and not as a big bang release.

Sorry I couldn't give any better news. But I will try and see if there isn't a way I can release smaller bits so that we can at least use the improved code-first APIs, Workflow Server & multi-server support.

**Comment by amattar** [comment]
Hi Sipke,

Thanks for the prompt update.

I believe it would be beneficial to have an intermediate version, say 1.5, and release whatever ready or needs less time to implement.

I think something like multi-server support is very crucial for many as I read other comments on other issues.

**Comment by mattbab** [comment]
@sfmskywalker Are the tasks to be completed for v2 documented in issues so the community can help you release sooner? That is assuming you're open to others contributing to the project.

**Comment by sfmskywalker** [comment]
An excellent question @mattbab . I didn't have a good overview for what I consider to be absolutely necessary for Elsa 2, but your question prompted me to get a start with that and collected them under a new [Elsa 2.0 MVP](https://github.com/elsa-workflows/elsa-core/milestone/8) milestone.

Help from the community is always more than welcome, and I invite anyone who would like to help to reach out to me for questions and discussion as needed.

Please take a look, and let me know what you think. Please keep in mind these are initial drafts, and we can refine them as necessary.

In summary, here's what I think needs to be done at a minimum:

- [x] Reimplement serialization
- [ ] ~~Finish GraphQL endpoint~~
- [ ] Finish REST API endpoint
- [ ] Refactor designer
- [ ] Implement standalone client-side SPA dashboard (that communicates with a configurable Elsa GraphQL endpoint)

With that in place, Elsa 2.0 will be on-par with Elsa 1 in terms of having a dashboard + designer. But with better code-first support, and a strong foundation to build the other features.

**Comment by xlegalles** [comment]
Hi Spike,
I just refresh this discussion to know what is the status of the v2: I can see from the Elsa 2.0 MVP milestone that you plan a version for march, is it still valid?
I also wonder about .NET Core 3.1: is it temporary to provide a .NET 5.0-only version of Elsa 2.0?
Thank you

**Comment by sfmskywalker** [comment]
@xlegalles So far the milestone of end of March seems still valid, yes.

About .NET Core 3.1 and .NET 5.0, I tried to stay compatible with .NET Core 3.1, but some dependencies require .NET 5, so it's proven to be tricky to retain. Due to limited bandwidth, I have to make choices on what I spend my time working on. Once Elsa 2 is merged into master and demand is high for .NET Core 3.1 support, we could of course consider creating a separate branch.

**Comment by mickdelaney** [comment]
Hi @sfmskywalker 

I am building a prototype right now in elsa 2x, i had been using the latest nuget packages **2.0.0-preview6.1002**
But I've noticed whilst looking in the _feature/elsa-2.0_ branch for guidance that there are differences between that branch and the published **2.0.0-preview6.1002** packages. 
I was considering just temporarily forking the repo and working in the _feature/elsa-2.0_ branch in the soluton itself, like adding some more samples to later throw away. 
What do you think ? 
Will the pre-release nugets be updated soon ? 
Is the branch unstable etc ? 

Let me know if I can be of any help. 
Thanks again.

**Reply by sfmskywalker** [reply]
Hi Mick,

You're right, the preview packages published to **NuGet** are way behind what's on the *feature/elsa-2.0* branch. The [MyGet feed](https://www.myget.org/gallery/elsa-2) on the other hand does contain the latest packages that reflect the tip. Because these contains the latest features and fixes, I typically create the following NuGet.config in the root of my solution:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear/>
    <add key="NuGet official package source" value="https://api.nuget.org/v3/index.json" />
    <add key="Elsa prereleases" value="https://www.myget.org/F/elsa-2/api/v3/index.json" />
  </packageSources>
</configuration>
```

Although I generally recommend working against these packages, I do realize it can be frustrating if not infuriating to have to deal with breaking changes (however small) when pulling latest versions. What I can do is push to NuGet on a slower cadence so that folks don't have to deal with breaking changes too frequently. Barring any unforeseen delays, the Elsa 2.0 stable release is slated for end of next month.

Let me know if you run into any issues.

**Reply by mickdelaney** [reply]
awesome thanks ;-)
